qualtran/bloqs/arithmetic.py

```diff
@@ -117,7 +169,7 @@ def signature(self):
         return Signature(
             [
                 Register("input", bitsize=self.bitsize, shape=(self.k,)),
-                Register("result", bitsize=2 * self.bitsize + 1),
+                Register("result", bitsize=2 * self.bitsize + 2),
```

Comment:
ditto

Reply:
So here was the problem. The original reference doesn't quote the bit count. So I must have gotten the +1 from adding two squares. In general the max number of bits needed goes like (k*(2**(n)-1)^2).bit_length() = 2n + ceil(log2(k)). So for the fusion example we got an extra 2 bits. I'll fix it for the general case.